base/arrayshow.jl

```diff
@@ -178,7 +178,7 @@ function print_matrix(io::IO, @nospecialize(X::AbstractVecOrMat),
     postsp = ""
     @assert textwidth(hdots) == textwidth(ddots)
     sepsize = length(sep)::Int
-    rowsA, colsA = UnitRange{Int}(axes(X,1)), UnitRange{Int}(axes(X,2))
+    rowsA, colsA = unitrange(axes(X,1)), unitrange(axes(X,2))
```

Comment:
Like this above, this probably normalizes us. Since `X` is `@nospecialize`d and called from places where you can't be sure about types, `unitrange` is not always going to be inferrable; the point of the `UnitRange{Int}` was to enforce the type even when the input types were not concretely known.

Again, we'll do it if we must.

Reply:
I don't see how the original version actually does what you want, wouldn't it need to be
```julia
rowsA, colsA = UnitRange{Int}(axes(X,1))::UnitRange{Int}, UnitRange{Int}(axes(X,2))::UnitRange{Int}
```
that is asserting the type explicitly? In any case, hopefully the `_print_matrix` change avoids this issue (in that type-inferrence is not important in `print_matrix` as `_print_matrix` knows the types).